docs/content/direct/architecture.md

```diff
@@ -466,5 +466,16 @@ The following section describes how transport controller works, while the descri
+
+The transport controller watches for `Binding` objects on the WDS, using an informer. Upon every add, update, and delete event from that informer, the controller puts a reference to that `Binding` object in its work queue. The transport controller also has informers on the inventory objects (both `ManagedCluster` and their associated `ConfigMap`) and on the wrapped objects (`ManifestWork`). Forked goroutines process items from the work queue. For a reference to a control or workload object, that processing starts with retrieving the informer's cached copy of that object. 
+
+The transport controller also maintains a finalizer on each Binding object. When processing a reference to a `Binding` object that no longer exists, the transport controller has nothing more to do --- the work was done earlier before removing that finalizer.
+
+When processing a reference to a `Binding` object that still exists, the transport controller looks at whether that `Binding` is in the process of being deleted. If so then the controller ensures that the corresponding wrapped object (`ManifestWork`) in the ITS no longer exists and then removes the finalizer from the `Binding`.
+
+When processing a `Binding` object that is not being deleted, the transport controller first ensures that the finalizer is on that object. Then the transport controller considers the `Binding`'s list of references to workload objects. The controller reads those objects from the WDS using a Kubernetes "dynamic" client. The controller goes through those objects one-by-one, applying template expansion for each destination if the object requests template expansion. If any of those objects requests template expansion and has a leaf string that actually involves template expansion: the controller accumulates a map from destination to slice of customized objects and then invokes the transport plugin on each of those slices, to ultimately produce a function from destination to wrapped object. If none of the selected workload objects actually involved any template expansion then the controller wraps the slice of workload objects to get one wrapped object and produces a constant function from destination to that one wrapped object. The controller then iterates over the `Binding`'s list of desinations and propagates the corresponding wrapped object (reported by the function just described) to the corresponding mailbox namespace.  Once the wrapped object is in the mailbox namespace of a cluster on the ITS, it's the agent responsibility to pull the wrapped object from there and apply/update/delete the workload objects on the WEC.
```

Comment:
1. On `leaf string`: having a leaf string implies having non-leaf strings. Both are not clear from this file alone.
2. This paragraph requires caching much information on the go, perhaps introducing bullets to better separate logic can help.

Reply:
I took out the word "leaf".

I factored that big paragraph.